src/worker.cpp

```diff
@@ -417,6 +417,8 @@ void Worker::setDeviceTreeAndJson()
                     ec.message());
             }
         }
+
+        m_isFactoryResetDone = true;
```

Comment:
will the above throw cause any issue to this value ? guess not , wanted your confirmation

Reply:
No, as of now, in case of throw above this place, we exit the service. Hence from our application point of view we have still not processed factory reset flow.